qatrfm/environment.py

```diff
@@ -64,37 +64,31 @@ def __init__(self, image, tf_file=None, num_domains=1,
         shutil.copy(self.tf_file, self.workdir + '/env.tf')
         os.chdir(self.workdir)
         self.domains = []
-        self.networks = []
-        self.networks.append(self.get_network())
-        self.logger.debug("Using network {} for the domains".
-                          format(self.networks[0]))
-        # TODO: check how many networks the user needs and fill this array
-        #       accordingly
+        self.net_octet = self.get_network_octet()
```

Comment:
I think this code apply's when running qa-trfm exec twice on a machine. If I'm not wrong this has a potential race in it.
If you run two qa-trfm at a time, both would get the same octet, cause linux-IP-stack isn't updated.

Some possibilities to solve:
1) Make a system wide mutex around every configure/lunch/apply 
2) Handle such things in a separate place as well (e.g. state file)
3) Not run qa-trfm in parallel for now.

I would prefere 1) or 3).

Reply:
Yes, and while testing it happened to me :)

I had to wait around 10 seconds to launch a second test to get the next octet... It is really race condition that I didn't want to solve in this PR. I was planning to bring that up for discussion but you are faster :)

Shall we go for 3) for now and implement 1) or 2) in a separate PR? We will need it if this enters production environments with automatic scheduling...

Reply:
Completely fine for me!